### ISSUE TITLE
0 loss when using ssd_pascal with my own dataset

### ISSUE THREAD

**Issue by Jack-lx-jiang** [weiliu89/caffe#707]
Hi,
When I used ssd to train my own dataset, I always got a 0 loss at the beginning and then the loss bounce around zero every iteration(sometimes even got a very small negative number! like -5e-8) Could some one explain why this happen? Doesn't my data set create correctly?  
![image](https://user-images.githubusercontent.com/19239472/29240238-840c6a4a-7f93-11e7-9fd4-4ec14802c47d.png)
![image](https://user-images.githubusercontent.com/19239472/29240281-78ca2022-7f94-11e7-9e22-6cc01fce65af.png)

**Comment by jtr2012** [comment]
@Jack-lx-jiang hi, have fixed your problem? I have the same problem.

**Comment by jtr2012** [comment]
@Jack-lx-jiang have you fixed the problem? I have the same problem with you.

**Comment by Jack-lx-jiang** [comment]
@jtr2012 Yes. In my case, the problem is that my ground truth boxes' coordinates are float then the reader cannot read them correctly.  To solve it, you can modify the XML reader to read float type data or you can just transfer your coordinate to Int.

**Comment by jtr2012** [comment]
@Jack-lx-jiang Really really thx! I'm just the same problem, and my training is going well now! You have just saved me! That problem confused me for two days!

**Comment by jtr2012** [comment]
@Jack-lx-jiang Have you met with such a kind problem only when testing?
![screenshot from 2017-10-22 16-51-23](https://user-images.githubusercontent.com/29769157/31871102-92856476-b7e4-11e7-9710-d9f9d754a947.png)

**Comment by Jack-lx-jiang** [comment]
@jtr2012 No, I haven't. What did you do to solve the problem??

**Comment by jtr2012** [comment]
@Jack-lx-jiang For the last problem, I just as you told transfer all the boundingbox data to int, and then everything seems well in training, but when in test period, this new problem came out, and the accuracy seems low, quite strange

**Comment by Jack-lx-jiang** [comment]
@jtr2012 You can track down from the 524th row of solver.cpp. I guess you may mistake your ground truth box's label. You can check both your data set's label and the data set's prototxt file. BTW, I just changed the reader to solve my problem.

**Comment by jtr2012** [comment]
@Jack-lx-jiang yes, I've already checked it, but nothing special for me. And could you tell me more how do you change the reader, change codes in which file, or something else?

**Comment by RoseLii** [comment]
please help me .
I download VOCtrainval_06-Nov-2007，but when i running ./data/VOC0712/create_list.sh have error that
y@y-System-Product-Name:~/2T/wangxingli/caffe$ ./data/VOC0712/create_list.sh
Create list for VOC2007 trainval...
cp: cannot stat '$/data/VOCdevkit/VOC2007/ImageSets/Main/trainval.txt': No such file or directory
sed: can't read /home/y/2T/wangxingli/caffe/data/VOC0712/trainval_img.txt: No such file or directory
sed: can't read /home/y/2T/wangxingli/caffe/data/VOC0712/trainval_img.txt: No such file or directory
cp: cannot stat '$/data/VOCdevkit/VOC2007/ImageSets/Main/trainval.txt': No such file or directory
sed: can't read /home/y/2T/wangxingli/caffe/data/VOC0712/trainval_label.txt: No such file or directory
sed: can't read /home/y/2T/wangxingli/caffe/data/VOC0712/trainval_label.txt: No such file or directory
paste: /home/y/2T/wangxingli/caffe/data/VOC0712/trainval_img.txt: No such file or directory
Create list for VOC2012 trainval...
cp: cannot stat '$/data/VOCdevkit/VOC2012/ImageSets/Main/trainval.txt': No such file or directory
sed: can't read /home/y/2T/wangxingli/caffe/data/VOC0712/trainval_img.txt: No such file or directory
sed: can't read /home/y/2T/wangxingli/caffe/data/VOC0712/trainval_img.txt: No such file or directory
cp: cannot stat '$/data/VOCdevkit/VOC2012/ImageSets/Main/trainval.txt': No such file or directory
sed: can't read /home/y/2T/wangxingli/caffe/data/VOC0712/trainval_label.txt: No such file or directory
sed: can't read /home/y/2T/wangxingli/caffe/data/VOC0712/trainval_label.txt: No such file or directory
paste: /home/y/2T/wangxingli/caffe/data/VOC0712/trainval_img.txt: No such file or directory
Create list for VOC2007 test...
cp: cannot stat '$/data/VOCdevkit/VOC2007/ImageSets/Main/test.txt': No such file or directory
sed: can't read /home/y/2T/wangxingli/caffe/data/VOC0712/test_img.txt: No such file or directory
sed: can't read /home/y/2T/wangxingli/caffe/data/VOC0712/test_img.txt: No such file or directory
cp: cannot stat '$/data/VOCdevkit/VOC2007/ImageSets/Main/test.txt': No such file or directory
sed: can't read /home/y/2T/wangxingli/caffe/data/VOC0712/test_label.txt: No such file or directory
sed: can't read /home/y/2T/wangxingli/caffe/data/VOC0712/test_label.txt: No such file or directory
paste: /home/y/2T/wangxingli/caffe/data/VOC0712/test_img.txt: No such file or directory
I1025 21:36:51.506589 1277 get_image_size.cpp:61] A total of 0 images.
Why is this wrong?

**Comment by dishita26** [comment]
> @jtr2012 Yes. In my case, the problem is that my ground truth boxes' coordinates are float then the reader cannot read them correctly. To solve it, you can modify the XML reader to read float type data or you can just transfer your coordinate to Int.

Hello,
I have transfer my co-ordinate to Int, still I am having a loss of zero. Do you have any other suggestions?
